hyperjumptech/monika
# Added: Secondary proof by checking to STUN server

# Monika Pull Request (PR)  
fix #310 

## What feature/issue does this PR add  
1. adding secondary proof by checking stun server to check if monika is connected to internet

## How did you implement / how did you fix it  
1.  using stun package that already used to obtain public IP
2. add `isConnectedToSTUNServer` as the status

## How to test  
1. run monika
2. see log that it is connected to stun server and getting public IP
3. disconnect your wifi, and it will log that monika cannot connect to stun server, and no probe is run
4. connect again your wifi, the probe should run as usual
<img width="1027" alt="Screen Shot 2021-08-27 at 15 44 08" src="https://user-images.githubusercontent.com/5974862/131107257-3b998b5c-5e34-495d-9ac1-a157867a6d6b.png">
